Mathlib/Geometry/Manifold/ContMDiff/Basic.lean

```diff
@@ -158,6 +158,18 @@ theorem contMDiffWithinAt_id : ContMDiffWithinAt I I n (id : M → M) s x :=
 
 end id
 
+/-! ### Iterated functions -/
+
+section Iterate
+
+/-- Iterated `ContMDiff` functions are `ContMDiff`. -/
+theorem ContMDiff.iterate {f : M → M} (fa : ContMDiff I I n f) (k : ℕ) :
+    ContMDiff I I n (f^[k]) := by
+  induction' k with k h; · simp only [Function.iterate_zero]; exact contMDiff_id
```

Comment:
I'd format this slightly differently:
- put each subgoal on its own line (the focusing dots are good and should stay)
- do you know about `simpa`? Your proofs can become `simpa only [Function.iterate_zero] using contMDiff_id` (and similarly for the other one). Unless they're a lot slower, `simpa using ...` is even better, the [style guide](https://leanprover-community.github.io/contribute/style.html#squeezing-simp-calls) explains why.